src/fireEvent.ts

```diff
@@ -14,8 +14,7 @@ const isTextInput = (element?: ReactTestInstance) => {
   // We have to test if the element type is either the TextInput component
   // (which would if it is a composite component) or the string
   // TextInput (which would be true if it is a host component)
-  // All queries but the one by testID return composite component and event
-  // if all queries returned host components, since fireEvent bubbles up
+  // All queries return host components but since fireEvent bubbles up
   // it would trigger the parent prop without the composite component check
   return (
     filterNodeByType(element, TextInput) ||
```

Comment:
1. Shouldn't this PR allow us to reduce this condition to only checking for `filterNodeByType(element, 'TextInput')`
2. Looks like this check should use detected host component name instead of hardcoded `TextInput`.

Reply:
Yes it should look for the detected host name for host textinput, I fixed this. Regarding removing the check on composite TextInput type I think it should remain since although all queries will now return host elements, fireEvent still triggers props on composite components by bubbling up and if we remove this check then the event will be triggered on the composite component TextInput even if it is non editable

Reply:
Makes sense.

I think we need to revisit our event handling system and figure out whether we can make it more robust, as there have been some issue with somewhat related code: #1261